django_project/certification/views/certificate.py

```diff
@@ -861,3 +862,94 @@ def preview_certificate(request, **kwargs):
         page.save()
 
     return response
+
+
+def issue_all_certificates(request, **kwargs):
+    """Issue all certificates to attendees in that course.
+
+     Kwargs:
+        project_slug (str): defines which project the attendee belongs.
+        course_slug (str): defines which course the attendee attended.
+        organisation_slug (str): defines which certifying organisation
+        conducted the course.
+
+    Returns:
+        String: Course attendees that are not yet certified.
+    """
+
+    project_slug = kwargs.pop('project_slug', None)
+    course_slug = kwargs.pop('course_slug', None)
+    organisation_slug = kwargs.get('organisation_slug', None)
+    course = Course.objects.get(slug=course_slug)
+    project = Project.objects.get(slug=project_slug)
+    certifying_organisation = \
+        CertifyingOrganisation.objects.get(slug=organisation_slug)
+
+    # Checking user permissions.
+    if request.user.is_staff or request.user == project.owner or \
+            request.user in project.certification_managers.all() or \
+            request.user in certifying_organisation.organisation_owners.all():
+        pass
+    else:
+        raise Http404
+
+    certified_attendees_pk = list(Certificate.objects.filter(
+            course=course).values_list('attendee', flat=True))
+
+    all_attendees_pk = list(CourseAttendee.objects.filter(
+            course=course).values_list(
+            'attendee', flat=True))
+
+    uncertified_attendees = []
+    for attendee in certified_attendees_pk:
+        if attendee in all_attendees_pk:
+            all_attendees_pk.remove(attendee)
+
+    for uncertified_attendee in all_attendees_pk:
+        uncertified_attendees.append(uncertified_attendee)
+
+    attendees = []
+    for pk in uncertified_attendees:
+        attendees.append(Attendee.objects.get(pk=pk))
+
+    pending_organisation = \
+        CertifyingOrganisation.objects.filter(approved=False)
+    has_pending = False
+    if pending_organisation:
+        has_pending = True
+
+    url = reverse('course-detail', kwargs={
+        'project_slug': project_slug,
+        'organisation_slug': organisation_slug,
+        'slug': course_slug
+    })
+
+    project_folder = (project.name.lower()).replace(' ', '_')
+    if request.method == 'POST':
+        site = request.get_host()
```

Comment:
__[Flake8]__

[F841] local variable 'site' is assigned to but never used


<!-- Comment by [SideCI](https://sideci.com) -->